Optimize rowwise_moments by MKL

xiaomengy opened the pull request:
Summary: Optimize rowwise_moments by MKL

Differential Revision: D12845220

BIT-silence commented:
Some benchmark.
LayerNormOp forward on CPU with size = [100, 100, 100, 100], axis = 1,
230ms -> 136ms

bddppq reviewed (approved):
Nice!

xiaomengy committed:
Optimize rowwise_moments by MKL (#13329)

Summary:
i-am-not-moving-c2-to-c10

Pull Request resolved: https://github.com/pytorch/pytorch/pull/13329

Optimize rowwise_moments by MKL

Reviewed By: houseroad

Differential Revision: D12845220

fbshipit-source-id: 9a8a3c5f970f12bd2a12398981b3fc034ff0fbd7
